p2p/gossip.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more detailg.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package p2p
+
+import (
+	log "github.com/ChainSafe/log15"
+	"github.com/libp2p/go-libp2p-core/network"
+)
+
+// gossip submodule
+type gossip struct {
+	host        *host
+	hasGossiped map[string]bool
+}
+
+// newGossip creates a new gossip instance from the host
+func newGossip(host *host) (g *gossip, err error) {
+	g = &gossip{
+		host:        host,
+		hasGossiped: make(map[string]bool),
+	}
+	return g, err
+}
+
+// handleMessage gossips messages that have not already been gossiped
```

Comment:
"handleMessage broadcasts messages that have not already been seen"?

Reply:
Not using "broadcast" and "seen" were also in line with keeping gossip separate as its own specific thing and "gossiped" because `hasGossiped`.

I can change `hasGossiped` to `hasBeenSeen` but we should probably also move where we set it to true since right now it is being set after the message has been gossiped.

Reply:
Ok, updated and simplified.

Reply:
I don't see why variable names need to match comments word for word. Broadcast is a word that concisely describes what is happening here in terms of propagation. Additionally, we are tracking the messages we have seen not the messages we have sent.

Reply:
Using broadcast here makes sense. I think that's just me being very literal with comments while still figuring out what exactly we need to what things we should keep separate.